apps/fz_http/lib/fz_http/saml/start_proxy.ex

```diff
@@ -60,7 +60,7 @@ defmodule FzHttp.SAML.StartProxy do
       |> set_service_provider()
       |> set_identity_providers(providers)
 
-    FzHttp.Config.put_env(:samly, Samly.Provider, samly_configs)
+    Application.put_env(:samly, Samly.Provider, samly_configs)
```

Comment:
Just to make sure I'm understanding -- this was changed so that you can override it in tests with `put_env_override/3` right?

Reply:
I did it because`FzHttp.Config.put_env/3` only writes to process dictionary in test env, so the Samly application will never be able to access the configuration we are setting here, so our tests are not reliable (testing broken behavior) while we can't even find the PID that holds that pdict to assert on the state in most of the cases.

> I wonder if there's some way to isolate the supervision tree per test process or maybe spin it different pieces of it selectively when testing certain modules.
You can't isolate it, but you can spin it up multiple times and test them. It's a lot of work to make every single place accept the process name, there are better and simpler ways - sandboxing supervision tree (for things like e2e tests) and testing processes individually in unit tests. 

Eg. we could have `start_proxy_test.exs` that will start a separate instance of a proxy and assert on the application state or Samly GenServer state.